### DISCUSSION TITLE
How to use this for generating video suggestions?

### DISCUSSION THREAD

**Discussion by anesuc** [imartinez/privateGPT#678] [Q&A]
Hi all,

I would like to feed this with JSON data that contains video ids, title, descriptions, views etc. I would like to then parse a video title or ID and tell it to return videos it thinks I should watch next based on the json data I gave it. Ideas on how to accomplish this?

**Comment by sime2408** [comment]
I am not sure can this extract multiple fields at once but you could try playing with it:

```python
from langchain.document_loaders import (
    JSONLoader,
)

LOADER_MAPPING = {
    ".json": (JSONLoader, {"jq_schema": ".videos[].description"}),
    # Add more mappings for other file extensions and loaders as needed
}
```
Few examples:
```
JSON        -> [{"id": ...}, {"title": ...}, {"description": ...}]
jq_schema   -> ".[].description"
        
JSON        -> {"videos": [{"id": ...}, {"title": ...}, {"description": ...}]}
jq_schema   -> ".videos[].title"

JSON        -> ["...", "...", "..."]
jq_schema   -> ".[]"
```
